### PR TITLE
FIO 7488: improve error handling

### PR DESCRIPTION
**Pull request by brendanbond**
## Link to Jira Ticket

https://formio.atlassian.net/browse/FIO-7488

## Description

Replaces `ValidatorError` and `DereferenceError` with a more generic `ProcessorError`
 
## Breaking Changes / Backwards Compatibility

n/a

## Dependencies

n/a

## How has this PR been tested?

n/a

## Checklist:

- [x] I have completed the above PR template
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if applicable)
- [x] My changes generate no new warnings
- [ ] My changes include tests that prove my fix is effective (or that my feature works as intended)
- [x] New and existing unit/integration tests pass locally with my changes
- [ ] Any dependent changes have corresponding PRs that are listed above

### CODE REVIEW

File: src/process/validation/rules/validateAvailableItems.ts [hunk 1]
```diff
@@ -42,23 +42,23 @@ async function getAvailableSelectValues(component: SelectComponent) {
             if (Array.isArray(component.data.values)) {
                 return mapStaticValues(component.data.values);
             }
-            throw new ValidatorError(
+            throw new Error(
```
Comment (AlexeyNikipelau) [inline src/process/validation/rules/validateAvailableItems.ts:45]: Why not use ProcessorError here?

Comment (brendanbond) [reply src/process/validation/rules/validateAvailableItems.ts:45]: whoopsie, fixed

File: src/process/validation/rules/validateMinimumSelectedCount.ts [hunk 2]
```diff
@@ -30,13 +30,13 @@ export const shouldValidate = (context: ValidationContext) => {
 
 function validateValue(value: DataObject[any]): asserts value is Record<string, boolean> {
     if (value == null || typeof value !== 'object') {
-        throw new ValidatorError(
+        throw new Error(
```
Comment (AlexeyNikipelau) [inline src/process/validation/rules/validateMinimumSelectedCount.ts:33]: And here

File: src/process/validation/rules/validateMaximumSelectedCount.ts [hunk 3]
```diff
@@ -30,13 +30,13 @@ export const shouldValidate = (context: ValidationContext) => {
 
 function validateValue(value: DataObject[any]): asserts value is Record<string, boolean> {
     if (value == null || typeof value !== 'object') {
-        throw new ValidatorError(
```
Comment (AlexeyNikipelau) [inline src/process/validation/rules/validateMaximumSelectedCount.ts:33]: Also here

**Review by AlexeyNikipelau — CHANGES_REQUESTED** [review]

**Pull request merged**